### PR TITLE
update docker config

### PR DESCRIPTION
**Pull request by maieutiquer**
This PR was inspired by #1499 and many changes were done to make it work under most circumstances.

Some of the changes include:
- :heavy_check_mark: adapt to new project structure in 13.x for SF4
- :heavy_check_mark: adjust commands as per the current Readme
- :heavy_check_mark: add github actions config to test that everything still works upon every PR
- :heavy_check_mark: `npm run webpack` issue is fixed in another PR and merged
- :heavy_check_mark: fixed SAML and assets errors
- :heavy_check_mark: fixed automatic user creation

Here's the result from deploying with `docker-compose up`:
http://env-9581639.jcloud-ver-jpc.ik-server.com/

Work that we still should do when we have time (probably in separate PRs?):
- the docker image should probably be built and published separately, we'll need an automated process for that using Github Actions
- multistage builds might help removing the installation of NodeJS inside the PHP-Apache image
- check Docker's best practices for persisting data and storing secrets
- locking dependencies of npm and composer for predictable builds - we'll need a separate action for that to avoid locking them by hand
- run Cypress in Github Actions (wait until Cypress is set up)
- test the supported range of versions in Github Actions, probably running different docker-compose.yml files (PHP7.2+NodeJS10+MySQL5.6, PHP7.3+NodeJS12+MySQL5.7, PHP7.4+NodeJS14+MySQL8.0)
- we may consider activating Dependabot for creating PRs with updated deps, once automated tests with Cypress are set up and the Travis build is fixed
- documentation should be provided, explaining how to do a first install, backups, updates, etc. including minimal supported Docker version (e.g. 19.3.0+) - where shall we do this, maybe in the Readme?

### CODE REVIEW

File: .docker/web/Dockerfile [hunk 1]
```diff
@@ -16,37 +17,38 @@ RUN apt-get update && apt-get install -y \
   wkhtmltopdf \
   xz-utils \
   xfonts-75dpi \
-  libav-tools \
+  ffmpeg \
```
Comment (LaurentGruber) [inline .docker/web/Dockerfile:20]: ffmpeg n'est plus nécessaire pour le moment

File: .docker/web/Dockerfile [hunk 2]
```diff
@@ -16,37 +17,38 @@ RUN apt-get update && apt-get install -y \
...
-
 RUN curl -sS https://getcomposer.org/installer | php
 RUN mv composer.phar /usr/local/bin/composer
 
-RUN wget http://download.gna.org/wkhtmltopdf/0.12/0.12.3/wkhtmltox-0.12.3_linux-generic-amd64.tar.xz
-RUN tar -xf wkhtmltox-0.12.3_linux-generic-amd64.tar.xz
+# RUN wget http://downloads.wkhtmltopdf.org/0.12/0.12.3/wkhtmltox-0.12.3_linux-generic-amd64.tar.xz
+# RUN tar -xf wkhtmltox-0.12.3_linux-generic-amd64.tar.xz
+RUN wget https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6-1/wkhtmltox_0.12.6-1.buster_amd64.deb
+RUN dpkg -i wkhtmltox_0.12.6-1.buster_amd64.deb
```
Comment (LaurentGruber) [inline .docker/web/Dockerfile:46]: wkhtmltopdf n'est plus nécessaire non plus

Comment (maieutiquer) [reply .docker/web/Dockerfile]: Merci, j'ai fait les deux changements et j'ai rebase le dernier 13.x, mais on obtient toujours la même erreur

**Comment by maieutiquer** [comment]
I managed to deploy 13.x when I commented out the following lines in three files inside the /vendor folder:
CredentialFactory.php:77
image
WebpackExtension.php:94
image
ThemeExtension.php:94
image

Here's the result: http://env-8813108.jcloud-ver-jpc.ik-server.com/

I don't know yet what the actual solution should be, I'm just sharing what worked after days of debugging so that we can start investigating

1. I'm not sure why `file_exists()` returns false when the file is there, should it be fixed or should we remove the check? What if the file is really missing for some reason?

2. How can we fix the lightsaml error caused by `empty($ownCredentials)`? Is it a vendor bug that should be reported to lightsaml or should we fix our default configuration for new installations somehow? Or should we use a shell command to update the configuration?

File: .docker/web/entrypoint.sh [hunk 3]
```diff
@@ -49,6 +53,9 @@ else
 fi
 
 echo "Setting correct file permissions"
-chmod -R 777 var/cache files/config var/log var/sessions files public/uploads
+chmod -R 777 var files config
```
Comment (LaurentGruber) [inline .docker/web/entrypoint.sh:56]: pour du dev c'est pas un souci, mais pour de la prod c'est un peu violent comme permission... Surtout sur config et files/config
Je remplacerais déjà par un chown www-data:www-data et du 755 partout voir 750 ?)

File: docker-compose.override.yml.dist [hunk 4]
```diff
@@ -1,42 +1,62 @@
...
+    container_name: claroline-web
+    build: .docker/web
     ports:
-      - 80:80
+      - "80:80"
     volumes:
-      - composer_cache:/root/.composer
-      - npm_cache:/root/.npm
-      - bower_cache:/root/.cache/bower
+      - .:/var/www/html/claroline # This is where the symfony files are stored
```
Comment (LaurentGruber) [inline docker-compose.override.yml.dist:9]: Pour du dev c'est très bien, pour de la prod il faudrait isoler les volumes plus finement.
Les données sur dans files qui doit donc avoir sont volume, il en faut également pour le cache, les logs et les sessions

File: docker-compose.override.yml.dist [hunk 5]
```diff
@@ -1,42 +1,62 @@
...
-      SECRET: ${SECRET}
-      ADMIN_FIRSTNAME: ${ADMIN_FIRSTNAME}
-      ADMIN_LASTNAME: ${ADMIN_LASTNAME}
-      ADMIN_USERNAME: ${ADMIN_USERNAME}
-      ADMIN_PASSWORD: ${ADMIN_PASSWORD}
-      ADMIN_EMAIL: ${ADMIN_EMAIL}
+      - APP_URL=${APP_URL}
+      - ENV=PROD
+      - APP_ENV=prod
+      - APP_DEBUG=0
```
Comment (LaurentGruber) [inline docker-compose.override.yml.dist:19]: Mettre des variables pour les 3 suivants pour pouvoir construire l'image en mode dev dans github action ...
ENV=PROD
APP_ENV=prod
APP_DEBUG=0

File: .docker/web/entrypoint.sh (hunk 3 above)
Comment (maieutiquer) [reply .docker/web/entrypoint.sh]: Corrigé:
```
chmod -R 750 var files config
chown -R www-data:www-data var files config
```

File: docker-compose.override.yml.dist (hunk 4 above)
Comment (maieutiquer) [reply docker-compose.override.yml.dist]: Je crois que je n'ai pas bien compris - j'ai regardé la documentation de Docker sur les Volumes et quelques tutoriels sur internet, et il semble que la plupart mettent le dossier du code source dans un volume.
Pour essayer, j'ai mis tous les fichiers du code source dans "claroline" (on peut aussi renommer si on trouve un meilleur nom - j'airais mis "app", mais je ne voulais pas qu'on se trompe avec l'ancien dossier "app" de la version 12.x)
Est-ce qu'on pourrait faire encore mieux ?

**Comment by Elorfin** [comment]
Sorry, your PR was mistakenly removed when we have merged `claroline/distribution` package and this one.

I need to fix your branch before merge. The conflicts resolution returned a strange result (because of the rewrite of the history).

**Comment by maieutiquer** [comment]
> Sorry, your PR was mistakenly removed when we have merged `claroline/distribution` package and this one.

That's OK, it's a good thing anyway, so I believe it's worth it

> I need to fix your branch before merge. The conflicts resolution returned a strange result (because of the rewrite of the history).

I have the diff in my own fork, I can use that to recreate the docker config. After that, we may discuss a few architectural decisions related to the Docker build in general, and to volumes in particular

**Comment by LaurentGruber** [comment]
remplacer par #1530

**Pull request closed without merging**